cli/azd/pkg/ioc/container.go

```diff
@@ -70,8 +70,8 @@ func (c *NestedContainer) RegisterNamedSingleton(name string, resolveFn any) err
 
 // Registers a resolver with a transient lifetime (instance per resolution)
 // Returns an error if the resolver is not valid
-func (c *NestedContainer) RegisterTransient(resolveFn any) error {
-	return c.inner.TransientLazy(resolveFn)
+func (c *NestedContainer) RegisterTransient(resolveFn any) {
```

Comment:
NIT: Rename this to `MustRegisterTransient`?

Reply:
This is really just an internal implementation detail that brings us inline with the same behavior that we exposed for `RegisterSingleton`.  If we fail to register the resolver we immediately panic because this is bad and shouldn't always be caught at dev time.

Reply:
I was not aware of this naming convention.  Do you have a link to a style guide or similar that we are adopting?

Reply:
> I was not aware of this naming convention. Do you have a link to a style guide or similar that we are adopting?

I don't think we have one.  This `Must` convention is used in a bit of the go standard library (e.g. regexp.MustCompile)

Reply:
I'm not sure. I haven't looked. I don't think `go` generally cares about how you should name things. After all, it's your application and you should choose what do with it. I think that's why Matt made it a `nit`.

The question is do we get value out of it naming it such a way? And I think the answer is yes.